code/modules/projectiles/guns/energy.dm

```diff
@@ -111,6 +111,13 @@
 	chambered = null //either way, released the prepared shot
 	newshot()
 
+/obj/item/gun/energy/kinetic_accelerator/process_chamber()

```

Comment:
it'd be better to overload the `robocharge` proc for kinetic accelerators. They're not supposed to use a borg's internal cell, anyway.

Also, why are you doing this here as opposed to under kinetic accelerators?

Reply:
Ok I'll look at overloading robocharge and making it so it doesn't pull power from their cell. 

My reasoning for putting it there was to just to have both `process_chamber` procs in the same place. I suppose I should put the overloaded robocharge proc in the KA file then.

Reply:
Wait, hold up. The `robocharge` proc is actually the problem here for the bug I'm trying to fix. If we let the proc run on the miner cyborg, it will recharge their KA prematurely and they will be able to spam it at the speed of other weapons like autorifles, revolvers, etc.

And if we're not using the proc to recharge their KA or pull from their battery, then the `robocharge` function would be useless to call, as it does nothing else.

Reply:
And that's precisely what will accomplish that.

```
/obj/item/gun/energy/kinetic_accelerator/robocharge()
	return
```

Now it doesn't draw from the internal cell and uses its own cell, like it should. It's no longer insanely faster and it no longer drains cyborgs after a couple of shots, which is what it's supposed to do.

Reply:
Ack, such a simple yet effective solution.. will change that